apis/network/v1beta1/ipset_webhook.go

```diff
@@ -74,3 +133,124 @@ func (r *IPSet) ValidateDelete() error {
+
+		// validate if requested network is in NetConfig
+		f := func(n Network) bool {
+			return n.Name == _net.Name
+		}
+		netIdx := slices.IndexFunc(netCfgSpec.Networks, f)
+
+		if netIdx >= 0 {
+			// validate if requested subnet is in NetConfig
+			f := func(n Subnet) bool {
+				return n.Name == _net.SubnetName
+			}
+			subNetIdx := slices.IndexFunc(netCfgSpec.Networks[netIdx].Subnets, f)
+
+			if subNetIdx >= 0 {
+				// net and subnet are valid
+				// validate the requested FixedIP
+				if _net.FixedIP != nil {
+					cidr := netCfgSpec.Networks[netIdx].Subnets[subNetIdx].Cidr
+					_, ipPrefix, ipPrefixErr := net.ParseCIDR(cidr)
+					if ipPrefixErr != nil {
+						// this should never happen as the subnet CIDR was already validated
+						// via the netcfg webhook
+						allErrs = append(allErrs, field.Invalid(path.Child("cidr"), cidr, errInvalidCidr))
+						return allErrs
+					}
+
+					path := path.Child("fixedIP")
+					if err := valiateAddress(*_net.FixedIP, ipPrefix, path); err != nil {
+						allErrs = append(allErrs, err...)
+					}
+				}
+
+				// check that there are not multiple have the defaultRoute flag
+				if _net.DefaultRoute != nil && *_net.DefaultRoute {
+					defaultRouteCount++
+
+					if defaultRouteCount > 1 {
+						allErrs = append(allErrs, field.Invalid(path.Child("defaultRoute"), _net.Name, errMultiDefaultRoute))
+					}
+				}
+
+				continue
+			}
+
+			// the requested subnet was not found in the network
+			allErrs = append(allErrs, field.Invalid(path.Child("subnetName"), _net.SubnetName, fmt.Sprintf(errSubnetNotInNetwork, _net.SubnetName, _net.Name)))
+			continue
+		}
+
+		// the requested net was not found in the NetConfig
+		allErrs = append(allErrs, field.Invalid(path.Child("name"), _net.Name, fmt.Sprintf(errNetworkNotFound, _net.Name)))
+	}
+
+	return allErrs
+}
+
+// valiateIPSetChanged
+// - if a previous requested network is still in the list
+// - if subnet changed within a network
+// - if fixedIP changed
+// - if defaultRoute changed
+func valiateIPSetChanged(
+	networks []IPSetNetwork,
+	oldNetworks []IPSetNetwork,
+	path *field.Path,
+) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for _netIdx, _net := range oldNetworks {
+		path := path.Child("networks").Index(_netIdx)
+
+		// validate if a previous requested network is still in the CR
+		f := func(n IPSetNetwork) bool {
+			return equality.Semantic.DeepEqual(n.Name, _net.Name)
+		}
+		netIdx := slices.IndexFunc(networks, f)
+		if netIdx < 0 {
+			// the network was removed
+			allErrs = append(allErrs, field.Invalid(path.Child("name"), _net.Name, fmt.Sprintf(errNetworkChanged, _net.Name)))
+			return allErrs
+		}
+
+		// validate if subnet changed within a network
+		if !equality.Semantic.DeepEqual(_net.SubnetName, networks[netIdx].SubnetName) {
+			allErrs = append(allErrs, field.Invalid(path.Child("subnetName"), _net.Name, fmt.Sprintf(errSubnetChanged, *&oldNetworks[_netIdx].SubnetName, *&networks[_netIdx].SubnetName)))
```

Comment:
Is there a reason that we use `*&` as opposed to just excluding that?

Reply:
no, fixing it

Reply:
done